addons/sale/models/account_move.py

```diff
@@ -154,6 +154,44 @@ def _is_downpayment(self):
         self.ensure_one()
         return self.line_ids.sale_line_ids and all(sale_line.is_downpayment for sale_line in self.line_ids.sale_line_ids) or False
 
+    def _get_amount_from_sales_order(self, order):
+        """
+        Consider all lines on any invoice in self that stem from the sales order `order`.
+        This function returns the sum of the totals of all those lines.
+        Note that this amount may be bigger than `order.amount_total`.
+        """
+        order_amount = 0
+        for invoice in self:
+            prices = sum(invoice.line_ids.filtered(lambda x: order in x.sale_line_ids.order_id).mapped('price_total'))
+            order_amount += invoice.currency_id._convert(
+                prices * -invoice.direction_sign,
+                order.currency_id,
+                invoice.company_id,
+                invoice.date,
+            )
```

Comment:
A better approach without an explicit conversion:
```suggestion
            price = sum(invoice.line_ids.filtered(lambda x: order in x.sale_line_ids.order_id).mapped('price_total'))
            rate = abs(invoice.amount_total / invoice.amount_total_signed) if invoice.amount_total_signed else 0.0
            price = invoice.company_currency_id.round(price)
```
But don't forget it will not be 100% accurate because you are cutting

Reply:
I agree in general.
But the function is used in `_compute_amount_to_invoice` to compute a stored field.
So I think it would be better not to change it (in stable at least)?
I basically just copied parts of `_compute_amount_to_invoice` so that I can use it somewhere else
(to make sure we compute it the same way).

Reply:
Ok then... even if I think it's a bad idea since the results is meaningless for the end-user